src/features/search/pages/DatesHoursModal.tsx

```diff
@@ -0,0 +1,366 @@
+import { RadioButton } from 'ui/components/radioButtons/RadioButton'
+import { Separator } from 'ui/components/Separator'
+import { VerticalUl } from 'ui/components/Ul'
+import { Close } from 'ui/svg/icons/Close'
+import { Spacer, Typo } from 'ui/theme'
+
+export enum RadioButtonDate {
+  TODAY = 'Aujourd’hui',
+  WEEK = 'Cette semaine',
+  WEEK_END = 'Ce week-end',
+  PRECISE_DATE = 'Date précise',
+}
+
+type DatesHoursModalFormData = {
+  hasSelectedDate: boolean
+  selectedDateChoice: DATE_FILTER_OPTIONS
+  selectedDate?: Date
+  hasSelectedHours: boolean
+  selectedHours?: Range<number>
+}
+
+type Props = {
+  title: string
+  accessibilityLabel: string
+  isVisible: boolean
+  hideModal: () => void
+}
+
+export const DATE_TYPES: Array<{
+  label: RadioButtonDate
+  type: DATE_FILTER_OPTIONS
+}> = [
+  { label: RadioButtonDate.TODAY, type: DATE_FILTER_OPTIONS.TODAY },
+  { label: RadioButtonDate.WEEK, type: DATE_FILTER_OPTIONS.CURRENT_WEEK },
+  { label: RadioButtonDate.WEEK_END, type: DATE_FILTER_OPTIONS.CURRENT_WEEK_END },
+  { label: RadioButtonDate.PRECISE_DATE, type: DATE_FILTER_OPTIONS.USER_PICK },
+]
+
+const MAX_HOUR = 24
+
+const DEFAULT_TIME_RANGE: Range<number> = [8, 22]
+
+const formatHour = (hour: number) => `${hour}h`
+
+const titleId = uuidv4()
+
+export const DatesHoursModal: FunctionComponent<Props> = ({
+  title,
+  accessibilityLabel,
+  isVisible,
+  hideModal,
+}) => {
+  const { navigate } = useNavigation<UseNavigationType>()
+  const { isDesktopViewport, modal } = useTheme()
+  const { searchState } = useSearch()
+  const logToggleDate = useLogFilterOnce(SectionTitle.Date)
+  const logToggleHour = useLogFilterOnce(SectionTitle.Hour)
+  const logSelectedDateChoice = useLogFilterOnce(SectionTitle.OfferDate)
+  const logChangeSelectedHours = useLogFilterOnce(SectionTitle.TimeSlot)
+  const [showCalendarPicker, setShowCalendarPicker] = useState<boolean>(false)
+  const { sliderLength } = useGetFullscreenModalSliderLength()
+
+  const TODAY = useMemo(() => new Date(), [])
+
+  const defaultValues = useMemo(() => {
+    return {
+      hasSelectedDate: !!searchState.date,
+      selectedDateChoice: searchState.date?.option ?? undefined,
+      selectedDate: searchState.date?.selectedDate
+        ? new Date(searchState.date?.selectedDate)
+        : undefined,
+      hasSelectedHours: !!searchState.timeRange,
+      selectedHours: searchState.timeRange ?? undefined,
+    }
+  }, [searchState.date, searchState.timeRange])
+
+  const {
+    handleSubmit,
+    reset,
+    setValue,
+    getValues,
+    formState: { isSubmitting, isValid, isValidating },
+    control,
+    watch,
+  } = useForm<DatesHoursModalFormData>({
+    mode: 'onChange',
+    resolver: yupResolver(datesHoursSchema),
+    defaultValues,
+  })
+  const watchedHasSelectedDate = watch('hasSelectedDate')
+  const watchedSelectedDate = watch('selectedDate')
+  const watchedHasSelectedHours = watch('hasSelectedHours')
```

Comment:
Je ne suis pas très fan des `watch` mais quand bien même, ici tu peux largement simplifier :
```suggestion
  const { hasSelectedDate, selectedDate, hasSelectedHours } = watch()
```

Si vraiment tu veux renommer tu peux faire 
```suggestion
  const { hasSelectedDate: watchedHasSelectedDate, selectedDate: watchedSelectedDate, hasSelectedHours: watchedHasSelectedHours } = watch()
```

Mais je suis même pas sûr que ce soit nécessaire.